workspaces/src/network/account.rs

```diff
@@ -17,6 +25,76 @@ impl Account {
     pub(crate) fn signer(&self) -> &InMemorySigner {
         &self.signer
     }
+
+    /// Call a contract on the network specified within `worker`, and return
+    /// a Builder object that we will make use to populate the rest of the
+    /// call details.
+    pub fn call<'a, T: Network>(
+        &self,
+        worker: &'a Worker<T>,
+        contract_id: AccountId,
+        function: &str,
+    ) -> CallBuilder<'a, T> {
+        CallBuilder::new(worker, contract_id, self.signer.clone(), function.into())
+    }
+
+    /// Transfer near to an account specified by `receiver_id` with the amount
+    /// specified by `amount_yocto`. Returns the execution details of this
+    /// transaction
+    pub async fn transfer_near<T: Network>(
+        &self,
+        worker: &Worker<T>,
+        receiver_id: AccountId,
+        amount_yocto: Balance,
+    ) -> anyhow::Result<CallExecutionDetails> {
+        worker
+            .transfer_near(self.signer(), receiver_id, amount_yocto)
+            .await
+    }
+
+    /// Deletes the current account, and returns the execution details of this
+    /// transaction. The beneciary will receive the funds of the account deleted
+    pub async fn delete_account<T: Network>(
+        self,
+        worker: &Worker<T>,
+        beneficiary_id: AccountId,
+    ) -> anyhow::Result<CallExecutionDetails> {
+        worker
+            .delete_account(self.id, &self.signer, beneficiary_id)
+            .await
+    }
+
+    /// Create a new sub account. Returns a CreateAccountBuilder object that
+    /// we can make use of to fill out the rest of the details. The sub account
+    /// id will be in the form of: "{new_account_id}.{parent_account_id}"
+    pub fn create_subaccount<'a, 'b, T: Network>(
+        &self,
+        worker: &'a Worker<T>,
+        new_account_id: &'b str,
+    ) -> CreateAccountBuilder<'a, 'b, T> {
+        CreateAccountBuilder::new(
+            worker,
+            self.signer.clone(),
+            self.id().clone(),
+            new_account_id,
+        )
+    }
+
+    pub async fn deploy<T: Network>(
+        self,
+        worker: &Worker<T>,
+        wasm: Vec<u8>,
```

Comment:
I wonder if, like the other conversion from `String` to `&str`, this should also be done because in the future we can assume that we don't actually require an owned vec to be able to serialize and send over the wire?

Reply:
That's the thing about a lot of the `near_primitives`/rpc, a lot of it requires owned types. Like `DeployContractAction` takes in a `Vec<u8>` so I can't do much here unless we change the deps

Reply:
Important meta note: whatever our *private* deps do internally should not have any bearing on the public API we expose. 

The RPC lib we use can use `&[u8]` or `Vec<u8>`, or `Bytes`, but that shouldn't change the public types. We should be able to refactor implementation from `Vec<u8>` to `Bytes` and back, if we like to, without making our user's change the code. 

In this sense, yeah, it seems that `wasm: &[u8]` would be a better signature here -- nothing in the problem fundamentally requires owned vec. I can also imagine the following idiom for sandbox testing: 

```
let CODE: &'static [u8] = include_bytes!("./target/wasm32-unknown-unknow/contrct.wasm");

#[test]
fn foo() {}
```

Reply:
Well, context is the king: in some cases, you want the abstraction to be inherently leaky to have, eg, maximal performance. What we are doing here is essentially a scripting layer, so we have a lot of freedom in how we organize internals, and have a lot of responsibility in making the public API most natural for the user.

Reply:
hmm, let's just do `wasm: AsRef<[u8]>` then so we can take in both `Vec<u8>` and `&[u8]`

Reply:
Using `AsRef` in the APIs is generally a bad idea, as it makes the function signature much harder to read, and also regresses type inference at the call-site. If the user have `xs: Vec<u8>`, they can just pass `&xs`. 

The only place where AsRef polymoprhism makes sense is for accepting a `&Path`. Note that it is needed there not to convert from owned to borrowed (PathBuf -> Path) but *between* borrowed types (&str -> &Path). And even than, it creates usability paper-cut, where passing in `PathBuf` works, but then you loose owenrship of it and can't, eg, print it in the error message.